File: src/main/java/com/tobe/healthy/lessonhistory/domain/LessonHistoryFiles.java
package com.tobe.healthy.lessonhistory.domain;

import static jakarta.persistence.FetchType.*;
import static jakarta.persistence.GenerationType.*;

import org.hibernate.annotations.DynamicUpdate;

import com.tobe.healthy.common.BaseTimeEntity;
import com.tobe.healthy.member.domain.Member;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.ToString;

@Entity
@DynamicUpdate
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@ToString
public class LessonHistoryFiles extends BaseTimeEntity<LessonHistoryFiles, Long> {

	@Id
	@GeneratedValue(strategy = IDENTITY)
	@Column(name = "lesson_history_files_id")
	private Long id;

	private String fileUrl;

	private int fileOrder;

	@ManyToOne(fetch = LAZY)
	@JoinColumn(name = "member_id")
	@ToString.Exclude
	private Member member;

	@ManyToOne(fetch = LAZY)
	@JoinColumn(name = "lesson_history_id")
	@ToString.Exclude
	private LessonHistory lessonHistory;

	@ManyToOne(fetch = LAZY)
	@JoinColumn(name = "lesson_history_comment_id")
	@ToString.Exclude
	private LessonHistoryComment lessonHistoryComment;

	public LessonHistoryFiles(String fileUrl, int fileOrder, Member member, LessonHistory lessonHistory) {
		this.fileUrl = fileUrl;
		this.fileOrder = fileOrder;
		this.member = member;
		this.lessonHistory = lessonHistory;
	}

	public LessonHistoryFiles(String fileUrl, int fileOrder, Member member, LessonHistory lessonHistory,
		LessonHistoryComment lessonHistoryComment) {
		this.fileUrl = fileUrl;
		this.fileOrder = fileOrder;
		this.member = member;
		this.lessonHistory = lessonHistory;
		this.lessonHistoryComment = lessonHistoryComment;
	}

	public static LessonHistoryFiles create(Member member, String fileUrl, int fileOrder) {
		return new LessonHistoryFiles(fileUrl, fileOrder, member, null);
	}

	public void updateFileOrder(int fileOrder) {
		this.fileOrder = fileOrder;
	}
}

File: src/main/kotlin/com/tobe/healthy/lesson_history/domain/entity/LessonHistoryFiles.kt
package com.tobe.healthy.lesson_history.domain.entity

import com.tobe.healthy.common.BaseTimeEntity
import com.tobe.healthy.member.domain.entity.Member
import jakarta.persistence.CascadeType.ALL
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.FetchType.LAZY
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType.IDENTITY
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne

@Entity
class LessonHistoryFiles(
    val fileUrl: String,

    val fileOrder: Int,

    @ManyToOne(fetch = LAZY)
    @JoinColumn(name = "member_id")
    val member: Member? = null,

    @ManyToOne(fetch = LAZY)
    @JoinColumn(name = "lesson_history_id")
    val lessonHistory: LessonHistory? = null,

    @ManyToOne(fetch = LAZY, cascade = [ALL])
    @JoinColumn(name = "lesson_history_comment_id")
    val lessonHistoryComment: LessonHistoryComment? = null,

    @Id
    @GeneratedValue(strategy = IDENTITY)
    @Column(name = "lesson_history_files_id")
    val id: Long = 0
) : BaseTimeEntity<LessonHistoryFiles, Long>() {

    companion object {
        fun create(member: Member, fileUrl: String, fileOrder: Int): LessonHistoryFiles {
            return LessonHistoryFiles(
                member = member,
                fileUrl = fileUrl,
                fileOrder = fileOrder
            )
        }
    }
}

